Documentation/project-docs/pullrequest-builds.md

```diff
@@ -0,0 +1,92 @@
+# Pull Request Builds
+
+As part of our Pull Requests we have some validation builds where we build the product and run tests on different Operative Systems and Build Configurations.
+
+### Build Matrix
+
+| Build | Framework | Builds Tests | Runs Tests |
+|:---|:---:|:---:|:---:|
+| Windows x86_Release | netcoreapp | X | X |
+| Windows x64_Debug | netcoreapp | X | X |
+| Windows NETFX_x86_Release | netfx | X | X |
+| Windows UWP_CoreCLR_x64_Debug | uap | X | X |
+| Windows UWP_NETNative_x86_Release | uapaot | X |   |
+| Linux x64_Release | netcoreapp | X | X |
+| Linux arm64_Release | netcoreapp | X | X |
+| Linux arm_Release | netcoreapp | X |   |
+| Linux musl_x64_Debug | netcoreapp | X |   |
+| MacOS x64_Debug | netcoreapp | X |   |
+| Packaging All Configurations x64_Debug | all | X | X |
+
+Our build definitions are defined by some .yml files with the following structure:
+
+- Entrypoint `.azure-pipelines.yml`
+- Shared build steps `eng/pipelines/corefx-base.yml`
+- Helix step `eng/pipelines/helix.yml`
+- Windows Matrix `eng/pipelines/windows.yml`
+- Linux Matrix `eng/pipelines/linux.yml`
+- MacOS Matrix `eng/pipelines/macos.yml`
+
+## Available builds
+
+- corefx-ci (Runs by default on every commit)
+- corefx-outerloop-linux
+- corefx-outerloop-osx
+- corefx-outerloop-windows
+
+### Azure pipelines commands
+
+```
+/azp help (displays help message)
+/azp run (runs all builds listed above)
+/azp run <build name> (/azp run corefx-outerloop-linux)
```

Comment:
@safern, could we create a corefx-outerloop such that I could type just `/azp run corefx-outerloop` and it would run all three of the OS-specific outerloops listed above?

Reply:
I'll investigate if this is possible. I'm not sure if we can achieve that, but we might be able to do so. I'll double check.

Reply:
In order to do this now, we would need to add a new dummy pipeline that does nothing, and then chain the other 3 pipelines to be triggered when that is finished. Are we OK with having a new pipeline that will always show green even though one of the others failed?

I will however talk to Azure DevOps folks, to maybe add a feature, to be able to specify a grouping name when triggering through comments or something like that.

Reply:
I just added a tracking issue for aliases/tags on a build, so that we don't have to do ugly workarounds.

Reply:
Ok, so I tried that workaround where we would add a dummy build and trigger the other 3 builds and the way this is structured in azure pipelines, won't actually work for PRs. What we could do is add an extra build (`corefx-outerloop`) which runs the whole outerloop matrix. If you want to just run osx, then you call `corefx-outerloop-osx` directly.